### DISCUSSION TITLE
Which IDE tool is used for iPXE? How to debug it?

### DISCUSSION THREAD

**Discussion by wonsch6207** [ipxe/ipxe#595] [Q&A]
I'm pretty sure that the OS environment is linux but which IDE tool is used? Eclipse? IntelliJ IDEA? VS Code?

I also would like to know how to debug the binary code line by line (not DBGC).

**Comment by NiKiZe** [comment]
Main environment is probably Emacs, running the code in qemu, or using the Linux build binary.

There is probably posts about this in the mailing list archives.
